### PR TITLE
added Property struct

### PR DESCRIPTION
**Pull request by cv711**

### CODE REVIEW

File: authres_test.go [hunk 1]
```diff
@@ -100,7 +127,8 @@ func TestParseAuthenticationResults(t *testing.T) {
 				t.Errorf("test %d: error: %v", n, err)
 			}
 			if res != nil && !reflect.DeepEqual(*res, tc.authres) {
-				t.Errorf("want: %+v, got: %+v", tc.authres, *res)
+				t.Errorf("want: %+v\n", tc.authres)
```
Comment (narqo) [inline authres_test.go]: I'd better leave `want: %v,\ngot: %v\n` here. It's an idiomatic way of doing assertion in tests.

File: authres_test.go [hunk 2]
```diff
@@ -26,7 +26,20 @@ func TestParseAuthenticationResults(t *testing.T) {
 					{
 						Method:     "spf",
 						Result:     "pass",
-						Properties: []string{"smtp+++mailfrom+++example.net"},
+						Properties: []Property{Property{"smtp", "mailfrom", "example.net"}},
```
Comment (narqo) [inline authres_test.go]: Not really an issue, but there is no need to duplicate `Property` here. Go's compiler is smart enough to guess the type of a slice's item.

**Review by narqo — COMMENTED** [review]
Looks good, but there is a merge conflict. Could you do a rebase?

**Comment by cv711** [comment]
Done!

**Pull request merged**